zio-http/src/main/scala/zio/http/codec/internal/EncoderDecoder.scala

```diff
@@ -324,34 +448,62 @@ private[codec] object EncoderDecoder                   {
     private def encodeBody(inputs: Array[Any]): Body = {
       if (isByteStream(flattened.content)) {
         Body.fromStream(inputs(0).asInstanceOf[ZStream[Any, Nothing, Byte]])
-      } else if (jsonEncoders.length == 0) Body.empty
-      else if (jsonEncoders.length == 1) {
-        val encoder = jsonEncoders(0)
+      } else {
+        if (inputs.length > 1) {
+          Body.fromMultipartForm(encodeMultipartFormData(inputs), formBoundary)
+        } else {
+          if (jsonEncoders.length == 0) Body.empty
+          else if (jsonEncoders.length == 1) {
+            val encoder = jsonEncoders(0)
+
+            encoder(inputs(0))
+          } else throw new IllegalStateException("A request on a REST endpoint should have at most one body")
+        }
+      }
+    }
 
-        encoder(inputs(0))
-      } else throw new IllegalStateException("A request on a REST endpoint should have at most one body")
+    private def encodeMultipartFormData(inputs: Array[Any]): Form = {
+      Form(
+        flattened.content.zipWithIndex.map { case (bodyCodec, idx) =>
+          val input = inputs(idx)
+          val name  = bodyCodec.name.getOrElse("field" + idx.toString)
+          bodyCodec match {
+            case BodyCodec.Multiple(schema, mediaType, _) if schema == Schema[Byte] =>
```

Comment:
This test is repeated, it could make sense to add `isBinaryStream` (and package-private `unsafeAsBinaryStream`) to `BodyCodec`.

Reply:
Ah it already exists I think.

Reply:
There is a similar one that I extracted, here I did not because the pattern match itself is also used to get some of the fields of `BinaryCodec.Multiple`